README.md

```diff
@@ -19,16 +19,28 @@ Once you have implemented the `fm` program (and tested it to the best of your ab
 
 ## Preprocessing
 
-*What preprocessing data do you store in files, and how?*
+In the preprocessing file we have 3 types of entries. The first entry marked by '>' is the name of the genome. The second entry is marked by '@' and it is the last column in the burrow-wheeler matrix, bwt. After this we write each entry from the C array, which is the accumulated buckets of the suffix array. Now each new entry is written on a new line in the file and is marked by '\*', where the first symbol following '\*' marks the symbol in the suffix array this is followed by an integer value indicating the starting position of the symbol in the suffix array.
```

Comment:
I hope you mean burrows-wheeler *transform* and not matrix, there? I don't look forward to loading $O(n^2)$ data in when running the tool.

Working with text files have pros and cons. The pros include that it is a lot easier to debug, because yourself and other tools can easily read them, and you can fix them if you discover an error. Cons include that they are slow to read, since you need at least a minimalistic parser. There is nothing wrong with using them, of course, and sometimes it is the right choice. For something like a serialised data structure, I would not choose them, though.

In Go, you have a serialisation framework, `gob`, that you could have used to write and read entire data structures. That won't be as fast as simply dumping data to a file, but implementing serialisation using it is a lot easier.